components/eam/src/chemistry/aerosol/wetdep.F90

```diff
@@ -225,101 +226,172 @@ subroutine clddiag(t, pmid, pdel, cmfdqr, evapc, &
    real(r8) sumpppr(pcols)       ! sum of positive precips from above
-   real(r8) cldv1(pcols)         ! precip weighted cloud fraction from above
-   real(r8) lprec                ! local production rate of precip (kg/m2/s)
+   real(r8) cldv1(pcols)         ! precip weighted cloud fraction from above (kg/m2/s)
    real(r8) lprecp               ! local production rate of precip (kg/m2/s) if positive
-   real(r8) rho                  ! air density
-   real(r8) vfall
-   real(r8) sumppr_cu(pcols)     ! Convective precipitation rate (kg/m2-s)
-   real(r8) sumpppr_cu(pcols)    ! Sum of positive convective precips from above
-   real(r8) cldv1_cu(pcols)      ! Convective precip weighted convective cloud fraction from above
-   real(r8) lprec_cu             ! Local production rate of convective precip (kg/m2/s)
-   real(r8) lprecp_cu            ! Local production rate of convective precip (kg/m2/s) if positive
-   real(r8) sumppr_st(pcols)     ! Stratiform precipitation rate (kg/m2-s)
-   real(r8) sumpppr_st(pcols)    ! Sum of positive stratiform precips from above
-   real(r8) cldv1_st(pcols)      ! Stratiform precip weighted stratiform cloud fraction from above
-   real(r8) lprec_st             ! Local production rate of stratiform precip (kg/m2/s)
-   real(r8) lprecp_st            ! Local production rate of stratiform precip (kg/m2/s) if positive
-   ! -----------------------------------------------------------------------
 
-   convfw = 1.94_r8*2.13_r8*sqrt(rhoh2o*gravit*2.7e-4_r8)
+
+   ! initiate variables
    do i=1,ncol
       sumppr(i) = 0._r8
       cldv1(i) = 0._r8
-      sumpppr(i) = 1.e-36_r8
-      sumppr_cu(i)  = 0._r8
-      cldv1_cu(i)   = 0._r8
-      sumpppr_cu(i) = 1.e-36_r8
-      sumppr_st(i)  = 0._r8
-      cldv1_st(i)   = 0._r8
-      sumpppr_st(i) = 1.e-36_r8
+      sumpppr(i) = 1.e-36_r8   ! not 0 because it will be divided
    end do
 
    do k = 1,pver
       do i = 1,ncol
-         cldv(i,k) = &
-            max(min(1._r8, &
-            cldv1(i)/sumpppr(i) &
-            )*sumppr(i)/sumpppr(i), &
-            cldt(i,k) &
-            )
-         lprec = pdel(i,k)/gravit &
-            *(prain(i,k)+cmfdqr(i,k)-evapr(i,k))
-         lprecp = max(lprec,1.e-30_r8)
-         cldv1(i) = cldv1(i)  + cldt(i,k)*lprecp
-         sumppr(i) = sumppr(i) + lprec
+         if (is_tot_cld) then 
+             cldv(i,k) = max(min(1._r8,cldv1(i)/sumpppr(i))*sumppr(i)/sumpppr(i),cld(i,k))
+         else
+             ! For convective and stratiform precipitation volume at the top interface of each layer. 
+             ! Neglect the current layer.
+             cldv(i,k) = max(min(1._r8,cldv1(i)/sumpppr(i))*(sumppr(i)/sumpppr(i)),0._r8)
+         endif
+         lprecp = max(lprec(i,k), 1.e-30_r8)
+         cldv1(i) = cldv1(i)  + cld(i,k)*lprecp
+         sumppr(i) = sumppr(i) + lprec(i,k)
+         sumppr_all(i,k) = sumppr(i)      ! save all sumppr to callrain_mix_ratio
          sumpppr(i) = sumpppr(i) + lprecp
+      enddo
```

Comment:
Not a big deal, but do we have a preference between `enddo` and `end do`, and `endif` and `end if`? It might be less confusing if we just select one or the other.

Reply:
I agree, let's stick with `enddo` and `endif`